pkg/controller/storagecluster/update.go

```diff
@@ -104,6 +115,15 @@ func (c *Controller) rollingUpdate(cluster *corev1.StorageCluster, hash string)
 				numUnavailableKvdb++
 			}
 		}
+
+		// If pxversion is greater than 2.13 then delete the portworx-api pods with csi-node-registrar containers as well when updating storage cluster
+		// This is done to re-register csi driver which gets removed when csi-node-driver-registrar is removed from px pods
+		if restartPxApiPods {
+			if apiPodName, ok := pxApiPodsMap[pod.Spec.NodeName]; ok {
+				oldPodsToDelete = append(oldPodsToDelete, apiPodName)
```

Comment:
Can we add UTs in controller_test.go to check for portworx-api pods in the deletion list?

Reply:
How to get the list of the portworx-api pods? It returns nil when I use k8sclient.